Animations/CMakeLists.txt

```diff
@@ -0,0 +1,220 @@
+        src/scripting/LuauLayer.cpp
+        src/scripting/GlobalApi.cpp
+        src/scripting/ScriptAnalyzer.cpp
+
+        src/audio/Audio.cpp
+        src/audio/WavLoader.cpp
+
+        src/editor/Timeline.cpp
+        src/editor/ProjectScreen.cpp
+        src/editor/ImGuiExtended.cpp
+        src/editor/EditorCameraController.cpp
+        src/editor/MenuBar.cpp
+        src/editor/EditorGui.cpp
+        src/editor/ImGuiTimeline.cpp
+        src/editor/DebugPanel.cpp
+        src/editor/SceneHierarchyPanel.cpp
+        src/editor/ExportPanel.cpp
+        src/editor/AnimObjectPanel.cpp
+        src/editor/EditorSettings.cpp
+        src/editor/ConsoleLog.cpp
+        src/editor/SceneManagementPanel.cpp
+        src/editor/AssetManagerPanel.cpp
+        src/editor/Gizmos.cpp
+
+        src/svg/Paths.cpp
+        src/svg/SvgCache.cpp
+        src/svg/Svg.cpp
+        src/svg/SvgParser.cpp
+
+        src/renderer/Shader.cpp
+        src/renderer/PerspectiveCamera.cpp
+        src/renderer/Renderer.cpp
+        src/renderer/Texture.cpp
+        src/renderer/Colors.cpp
+        src/renderer/Fonts.cpp
+        src/renderer/OrthoCamera.cpp
+        src/renderer/GLApi.cpp
+        src/renderer/Framebuffer.cpp
+
+        src/utils/CMath.cpp
+        src/utils/TableOfContents.cpp
+
+        src/math/DataStructures.cpp
+
+        src/platform/FileSystemWatcherWin32.cpp
+        src/platform/PlatformLinux.cpp
+        src/platform/FileSystemWatcherLinux.cpp
+        src/platform/PlatformWin32.cpp
+
+        src/core/ImGuiLayer.cpp
+        src/core/Input.cpp
+        src/core/ProjectApp.cpp
+        src/core/Application.cpp
+        src/core/GladLayer.cpp
+        src/core/Window.cpp
+
+        src/latex/LaTexLayer.cpp
+
+        src/SingleIncludeImpls.cpp
+        src/core.cpp
+        src/main.cpp)
+
+if(MATH_ANIMATION_OS_LINUX)
+    set(MATH_ANIMATIONS_PLATFORM_SOURCE
+            src/platform/PlatformLinux.cpp
+            src/platform/FileSystemWatcherLinux.cpp)
+
+    # TODO: The source code used to use _MAX_PATH, but this can cause issues on Linux
+    # see: https://stackoverflow.com/questions/833291/is-there-an-equivalent-to-winapis-max-path-under-linux-unix
+    set(MATH_ANIMATIONS_MAX_PATH 4096)
+elseif(MATH_ANIMATION_OS_WINDOWS)
+    set(MATH_ANIMATIONS_PLATFORM_SOURCE
+            src/platform/FileSystemWatcherWin32.cpp
+            src/platform/PlatformWin32.cpp)
+
+    # Alias to the macro _MAX_PATH on windows
+    set(MATH_ANIMATIONS_MAX_PATH _MAX_PATH)
+else()
+    message(FATAL_ERROR "Targeting unsupported operating system ${CMAKE_SYSTEM_NAME}")
+endif()
+
+add_executable(MathAnimations ${MATH_ANIMATIONS_SOURCE} ${MATH_ANIMATIONS_PLATFORM_SOURCE})
+target_include_directories(MathAnimations PUBLIC ${MATH_ANIMATIONS_INCLUDE_DIR})
+target_compile_definitions(MathAnimations PUBLIC
+        MATH_ANIMATIONS_MAX_PATH=${MATH_ANIMATIONS_MAX_PATH}
+)
+target_compile_definitions(MathAnimations PRIVATE
+        $<$<CONFIG:Debug>:_DEBUG=1>
```

Comment:
I see that you've added these definitions in here but changed the `#ifdef`s in the code to use `NDEBUG`. Is there any reason for that or can we revert those changes since the definitions are being added here?

Reply:
Oversight on my end, changes reverted.